src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -600,12 +601,28 @@ func (c *controller) UploadLogs(ctx context.Context, wg *sync.WaitGroup, status
 				}
 				podName = pods[0].Name
 			}
-			err := common.UploadPodLogs(c.kc, c.ic, c.ClusterID, podName, c.Namespace, controllerLogsSecondsAgo, c.log)
+			var err error
+			if errorDetected != status.HasError() {
+				//once error is detected in the flow, upload the full capacity of debugging logs
+				//this is required only once, upon the detection of a fatal error
+				errorDetected = status.HasError()
+				_ = c.uploadSummaryLogs(podName, c.Namespace, controllerLogsSecondsAgo, errorDetected)
```

Comment:
Please check the return error and log if necessary. Errors should never pass silently

Reply:
it must in this case, because we have nothing to do with it. The controller is about to die, the logs will never be uploaded again so there is no point to log anything here. Inside the flow there are ample logs

Reply:
Logic changed. This remark was resolved as part of the new flow